Use gtk_show_uri() instead of deprecated gnome-open.
This also fixes the issue that the error dialog is never shown, when an error
occurs while openning the URL, because gnome-open was run successfully. Thus
gdk_spawn_command_line_on_screen() always returns TRUE, even if gnome-open actually
fails to open the URL, e.g due to missing browser.

--- a/network-manager-applet-0.8.1/src/applet-dialogs.c	2012-02-22 17:36:24.368509502 +0100
+++ b/network-manager-applet-0.8.1/src/applet-dialogs.c	2012-03-01 10:36:48.898968507 +0100
@@ -678,29 +678,20 @@ about_dialog_handle_url_cb (GtkAboutDial
 {
 	GError *error = NULL;
 	gboolean ret;
-	char *cmdline;
 	GdkScreen *gscreen;
 	GtkWidget *error_dialog;
 
 	gscreen = gtk_window_get_screen (GTK_WINDOW (about));
 
-	cmdline = g_strconcat ("gnome-open ", url, NULL);
-	ret = gdk_spawn_command_line_on_screen (gscreen, cmdline, &error);
-	g_free (cmdline);
-
-	if (ret == TRUE)
-		return;
-
-	g_error_free (error);
-	error = NULL;
-
-	cmdline = g_strconcat ("xdg-open ", url, NULL);
-	ret = gdk_spawn_command_line_on_screen (gscreen, cmdline, &error);
-	g_free (cmdline);
-	
+	ret = gtk_show_uri (gscreen, url, gtk_get_current_event_time (), &error);
+
 	if (ret == FALSE) {
-		error_dialog = gtk_message_dialog_new ( NULL, GTK_DIALOG_MODAL, GTK_MESSAGE_INFO, GTK_BUTTONS_OK, "Failed to show url %s", error->message); 
+		error_dialog = gtk_message_dialog_new (NULL, GTK_DIALOG_MODAL | GTK_DIALOG_DESTROY_WITH_PARENT,
+		                                       GTK_MESSAGE_INFO, GTK_BUTTONS_OK, "%s",
+		                                       error->message);
+		gtk_window_set_title (GTK_WINDOW (error_dialog), "Failed to show URL");
 		gtk_dialog_run (GTK_DIALOG (error_dialog));
+		gtk_widget_destroy (error_dialog);
 		g_error_free (error);
 	}
 
@@ -718,23 +708,17 @@ about_dialog_handle_email_cb (GtkAboutDi
 
 	gscreen = gtk_window_get_screen (GTK_WINDOW (about));
 
-	cmdline = g_strconcat ("gnome-open mailto:", email_address, NULL);
-	ret = gdk_spawn_command_line_on_screen (gscreen, cmdline, &error);
+	cmdline = g_strconcat ("mailto:", email_address, NULL);
+	ret = gtk_show_uri (gscreen, cmdline, gtk_get_current_event_time (), &error);
 	g_free (cmdline);
 
-	if (ret == TRUE)
-		return;
-
-	g_error_free (error);
-	error = NULL;
-
-	cmdline = g_strconcat ("xdg-open mailto:", email_address, NULL);
-	ret = gdk_spawn_command_line_on_screen (gscreen, cmdline, &error);
-	g_free (cmdline);
-	
 	if (ret == FALSE) {
-		error_dialog = gtk_message_dialog_new ( NULL, GTK_DIALOG_MODAL, GTK_MESSAGE_INFO, GTK_BUTTONS_OK, "Failed to show url %s", error->message); 
+		error_dialog = gtk_message_dialog_new (NULL, GTK_DIALOG_MODAL | GTK_DIALOG_DESTROY_WITH_PARENT,
+		                                       GTK_MESSAGE_INFO, GTK_BUTTONS_OK, "%s",
+		                                       error->message);
+		gtk_window_set_title (GTK_WINDOW (error_dialog), "Failed to open e-mail");
 		gtk_dialog_run (GTK_DIALOG (error_dialog));
+		gtk_widget_destroy (error_dialog);
 		g_error_free (error);
 	}
 }
@@ -777,7 +760,7 @@ applet_about_dialog_show (NMApplet *appl
 
 	gtk_show_about_dialog (NULL,
 	                       "version", VERSION,
-	                       "copyright", _("Copyright \xc2\xa9 2004-2008 Red Hat, Inc.\n"
+	                       "copyright", _("Copyright \xc2\xa9 2004-2012 Red Hat, Inc.\n"
 					                  "Copyright \xc2\xa9 2005-2008 Novell, Inc."),
 	                       "comments", _("Notification area applet for managing your network devices and connections."),
 	                       "website", "http://www.gnome.org/projects/NetworkManager/",
